apps/coordinator/src/clients/fees.ts

```diff
@@ -0,0 +1,97 @@
+import { BlockchainClient } from "@caravan/clients";
+import { useQuery } from "@tanstack/react-query";
+import { useGetClient } from "hooks/client";
+
+export enum FeePriority {
+  HIGH = "HIGH",
+  MEDIUM = "MEDIUM",
+  LOW = "LOW",
+}
+
+/**
+ * Confirmation targets for fee estimation in blocks
+ *
+ * These values align with common confirmation targets used generally:
+ * - HIGH: Next block (10 minutes)
+ * - MEDIUM: ~3 blocks (30 minutes)
+ * - LOW: ~6 blocks (1 hour)
+ *
+ * @see https://gist.github.com/morcos/d3637f015bc4e607e1fd10d8351e9f41
+ */
+export const CONFIRMATION_TARGETS = {
+  [FeePriority.HIGH]: 1, // Next block (~10 min)
+  [FeePriority.MEDIUM]: 3, // Within Next 3 blocks ~30 min
+  [FeePriority.LOW]: 6, // Within Next 6 blocks ~1 hour
+};
+
+/**
+ * Default fee rates for different priority levels (in sat/vB)
+ * Used as fallback when blockchain client fails
+ *
+ * Fallback values based on :
+ * https://b10c.me/blog/003-a-list-of-public-bitcoin-feerate-estimation-apis/
+ * These values are reasonable defaults but will be less accurate
+ */
+const DEFAULT_FEE_RATES = {
+  [FeePriority.HIGH]: 32.75,
+  [FeePriority.MEDIUM]: 32.75,
+  [FeePriority.LOW]: 20.09,
+};
+
+const feeEstimateKeys = {
+  all: ["fees"] as const,
+  feeEstimate: (priority: FeePriority) =>
+    [...feeEstimateKeys.all, priority] as const,
+};
+
+const useGetFeeEstimate = async (
```

Comment:
given a specific priority, get the fee estimate. since there are only three possibilities, this lets use the query cache effectively.